azure-kusto-data/azure/kusto/data/request.py

```diff
@@ -75,15 +100,31 @@ def __init__(self, connection_string):
 
         for kvp_string in connection_string.split(";"):
             key, _, value = kvp_string.partition("=")
-            self[key] = value
+            keyword = self.ValidKeywords.parse(key)
+            if keyword.is_str_type():
+                self[keyword] = value.strip()
+            if keyword.is_bool_type():
+                if value.strip() in ["True", "true"]:
+                    self[keyword] = True
+                elif value.strip() in ["False", "false"]:
+                    self[keyword] = False
+                else:
+                    raise KeyError("Expected aad federated security to be bool. Recieved %s" % value)
 
     def __setitem__(self, key, value):
         try:
             keyword = key if isinstance(key, self.ValidKeywords) else self.ValidKeywords.parse(key)
         except KeyError:
             raise KeyError("%s is not supported as an item in KustoConnectionStringBuilder" % key)
 
-        self._internal_dict[keyword] = value if keyword is self.ValidKeywords.aad_federated_security else value.strip()
+        if keyword.is_str_type():
+            self._internal_dict[keyword] = value.strip()

```

Comment:
this will throw if value is None, should we raise `ValueError` for  `None`? maybe as part of `ensure_type`?